zerver/lib/markdown/__init__.py

```diff
@@ -1605,6 +1618,8 @@ def url_to_a(
     href = rewrite_local_links_to_relative(db_data, href)
 
     a.set("href", href)
+    if is_previewable(href):
+        a.set("class", "previewable")
```

Comment:
The results of backend Markdown processing are saved to the database and fixed for all time, so this will prevent us from adding or removing `previewable` in historical messages. This should be done on the frontend, not the backend.